src/project/projectSlice.ts

```diff
@@ -15,42 +15,100 @@
  * along with this program. If not, see https://www.gnu.org/licenses/.
  */
 
-import { createSlice } from '@reduxjs/toolkit';
+import { createAction, createSlice } from '@reduxjs/toolkit';
+import { setUsers, User } from 'terraso-client-shared/account/accountSlice';
+import {
+  Membership,
+  setMembers,
+} from 'terraso-client-shared/memberships/membershipsSlice';
 import * as projectService from 'terraso-client-shared/project/projectService';
-import { createAsyncThunk } from 'terraso-client-shared/store/utils';
+import { setSites, Site } from 'terraso-client-shared/site/siteSlice';
+import {
+  createAsyncThunk,
+  dehydrated,
+} from 'terraso-client-shared/store/utils';
+
+const { plural: dehydrateProjects, sing: dehydrateProject } = dehydrated<
+  Project,
+  HydratedProject
+>({
+  memberships: setMembers,
+  users: setUsers,
+  sites: setSites,
+});
+
+export type SerializableSet = Record<string, boolean>;
```

Comment:
This is to overcome the fact that you can't store the Set object in the Redux store. Instead, an object's presence is set by setting the object key in the SerializableSet.